### PR TITLE
Add updates to complete Tests for Install, Save, Publish, Get-Installed, and Uninstall tests.

### PR DESCRIPTION
**Pull request by alerickson**
# PR Summary

Adding updates to tests, specifically focusing on Install-PSResource, Save-PSResource, Publish-PSResource, Get-InstalledPSResource, and Uninstall-PSResource tests. These tests are also validated against Ubuntu 20.04. 
Uninstall now also has an out ref for error so that all errors get written out from the top level calling method.  This is to prevent output of multiple errors (which can be confusing and is unnecessary).

## PR Context

Tests within Uninstall and Get-InstalledPSResource that were previously commented out have been uncommented, updated, and validated.  There are some updates for Linux tests and changes made to ensure proper testing on both Windows and Linux.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

**Review by PaulHigin — APPROVED** [review]
LGTM

File: test/GetInstalledPSResource.Tests.ps1 [hunk 1]
```diff
@@ -100,8 +103,7 @@ Describe 'Test Get-InstalledPSResource for Module' {
     }
 
     It "Get resources when given Name, and Version is '*'" {
-        $pkgs = Get-InstalledPSResource -Name Carbon -Version "*"
+        $pkgs = Get-InstalledPSResource -Name ContosoServer -Version "*"
         $pkgs.Count | Should -BeGreaterOrEqual 2
     }
 }
-#>
```
Comment (anamnavi) [inline test/GetInstalledPSResource.Tests.ps1:107]: nitpick, add newline :)

File: test/UninstallPSResource.Tests.ps1 [hunk 2]
```diff
@@ -10,180 +9,147 @@ Describe 'Test Uninstall-PSResource for Modules' {
...
+    It "Uninstall module that is a dependency for another module using -Force" {
+        $null = Install-PSResource "test_module" -Repository $TestGalleryName -TrustRepository -WarningAction SilentlyContinue
+
+        $res = Uninstall-PSResource -Name "RequiredModule1" -Force
+        
+        $pkg = Get-Module "RequiredModule1" -ListAvailable
+        $pkg | Should -Be $null
     }
 }
-#>
```
Comment (anamnavi) [inline test/UninstallPSResource.Tests.ps1:189]: nitpick, add newline :)

**Review by anamnavi — APPROVED** [review]
LGTM

File: test/UninstallPSResource.Tests.ps1 [hunk 3]
```diff
@@ -10,180 +9,147 @@ Describe 'Test Uninstall-PSResource for Modules' {
...
-        $res = Uninstall-PSResource -Name "ActiveDirectoryTools" -WhatIf
+    It "Uninstall module that is a dependency for another module using -Force" {
+        $null = Install-PSResource "test_module" -Repository $TestGalleryName -TrustRepository -WarningAction SilentlyContinue
+
+        $res = Uninstall-PSResource -Name "RequiredModule1" -Force
+        
+        $pkg = Get-Module "RequiredModule1" -ListAvailable
+        $pkg | Should -Be $null
     }
 }
```
Comment (anamnavi) [inline test/UninstallPSResource.Tests.ps1:155]:
```suggestion
}


```

Comment (anamnavi) [inline test/UninstallPSResource.Tests.ps1:155]:
```suggestion
}

```

File: test/GetInstalledPSResource.Tests.ps1 [hunk 4]
```diff
@@ -100,8 +103,7 @@ Describe 'Test Get-InstalledPSResource for Module' {
     }
 
     It "Get resources when given Name, and Version is '*'" {
-        $pkgs = Get-InstalledPSResource -Name Carbon -Version "*"
+        $pkgs = Get-InstalledPSResource -Name ContosoServer -Version "*"
         $pkgs.Count | Should -BeGreaterOrEqual 2
     }
 }
```
Comment (anamnavi) [inline test/GetInstalledPSResource.Tests.ps1:109]:
```suggestion
}

```

**Pull request merged**